docs/src/pages/theming.mdx

```diff
@@ -0,0 +1,430 @@
+  fileTypes: string[];
+};
+
+type DropzoneCallbackArguments = {
+  ready: boolean;
+  isUploading: boolean;
+  uploadProgress: number;
+  fileTypes: string[];
+  isDragActive: boolean;
+};
+```
+
+## How to theme
+
+### With TailwindCSS
+
+#### Configuring TailwindCSS
+
+To leverage best developer experience, we strongly recommend extending your
+TailwindCSS config using our utility function `withUt`
+
+<Callout type="warning">
+  Note: `withUt` is optional but if you do not use it, you will not be able to
+  use new classes and variants it adds. Also, in case you are not using it, make
+  sure to import default stylesheet into your app.
+</Callout>
+
+```ts filename="tailwind.config.ts"
+import { withUt } from "@uploadthing/react/tw";
+
+/** @type {import('tailwindcss').Config} */
+export default withUt({
+  // your config goes here
+});
+```
+
+```js filename="tailwind.config.js"
+// @ts-check
+const { withUt } = require("@uploadthing/react/tw");
+
+/** @type {import('tailwindcss').Config} */
+module.exports = withUt({
+  // your config goes here
+});
+```
+
+<Callout type="warning">
+  Note: Once you extended your config, it is recommended to remove default
+  stylesheet import
+</Callout>
+
+`withUt` does two major things:
+
+- Adds new entry to `content` property so that it includes UploadThing's
+  components. It allows to avoid having duplicated styles in your bundle.
+- Registers plugin and adds nice-to-have classes to improve styling experience.
+
+#### `className` prop
+
+`className` prop accepts any classes so you can pass there anything you like.
+When it comes to TailwindCSS, to make styling easier, `withUt` adds few utility
+classes and variants for you to leverage.
+
+Tailwind variants that allow applying styles to internal elements and their
+arbitrary variant alternatives:
+
+| Custom variant       |                        Arbitrary variant |
+| :------------------- | ---------------------------------------: |
+| `ut-button`          |          `&>*[data-ut-element="button"]` |
+| `ut-allowed-content` | `&>*[data-ut-element="allowed-content"]` |
+| `ut-label`           |           `&>*[data-ut-element="label"]` |
+| `ut-upload-icon`     |     `&>*[data-ut-element="upload-icon"]` |
+
+Tailwind classes that allow applying styles based on state of element and their
+arbitrary `data` class alternatives:
+
+| Custom class   |          Arbitrary class |
+| :------------- | -----------------------: |
+| `ut-readying`  |  `data-[state=readying]` |
+| `ut-ready`     |     `data-[state=ready]` |
+| `ut-uploading` | `data-[state=uploading]` |
+
+These variants and classes can be used in conjunction with each other to make
+component match your design in the exact way you want.
+
+```jsx
+<UploadButton
+    className="bg-blue-500 ut-button:bg-cyan-500 ut-button:ut-readying:bg-cyan-500/50"
+               ^           ^ applied to button   ^ applied to button when loading config
+               | applied to container
+
+/>
```

Comment:
Very nice. We could also render this component beneath to show the result

Reply:
Makes sense, will do

Reply:
Would screenshots be enough or you want to render the component within the markdown?

Reply:
> Would screenshots be enough or you want to render the component within the markdown?

Adding @uploadthing/react as a dep and just render it would probably be the best.

Reply:
@juliusmarminge But doesn't FE component require API endpoint to be present to pull configuration from? I mean I'm not sure If I can define API endpoint in Nextra

Reply:
Nextra is just Nextjs, so you could.